packages/btcindexer/src/btcindexer.ts

```diff
@@ -144,6 +151,81 @@ export class Indexer {
 		return deposits;
 	}
 
+	async processFinalizedTransactions(): Promise<void> {
+		const finalizedTxs = await this.d1
+			.prepare("SELECT tx_id, block_hash FROM nbtc_txs WHERE status = 'finalized'")
+			.all<{ tx_id: string; block_hash: string }>();
+
+		if (!finalizedTxs.results || finalizedTxs.results.length === 0) {
+			return;
+		}
+
+		const updates: D1PreparedStatement[] = [];
+		// TODO: do we imidietly process it and check if it was succesful and just change the status to minted? This should be a matter of seconds at most.
```

Comment:
that should be checked later, at other stage:
- either after collecting the Sui tx results
- in other loop. 

Let's keep the TODO, and create a task.